### PR TITLE
New mark repair

### PR DESCRIPTION
**Pull request by thelord82**
POST works with input text, index.html has what it needs to upload a file. Not implemented, not working. No CGI in this one yet.

### CODE REVIEW

File: src/http.cpp
Comment (francislah) [inline src/http.cpp:1]: Buffer is what 🤔?

**Review by francislah — APPROVED** [review]
Good

File: include/server.hpp [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 #include <map>
 #include <string>
+#include <sys/types.h>
+#include <dirent.h>
```
Comment (francislah) [inline include/server.hpp:6]: Is it allowed by the pdf?

File: src/server.cpp [hunk 2]
```diff
@@ -250,10 +250,140 @@ bool HttpServer::deleteMethod(HttpRequest &request, HttpResponse &response,
...
 
+
+//!LOCAL FOR NOW - TESTING IN PROGRESS
+std::string extractValue(const std::string& data, const std::string& start, const std::string& end) {
+    size_t startPos = data.find(start) + start.length();
+    size_t endPos = data.find(end, startPos);
+    if (end.empty()) {
+        endPos = data.length();
+    }
+    return data.substr(startPos, endPos - startPos);
```
Comment (francislah) [inline src/server.cpp:261]: Segfault lorsque fichier trop grand

**Review by francislah — CHANGES_REQUESTED** [review]

File: src/server.cpp (hunk 2 above)
Comment (thelord82) [reply src/server.cpp:261]: Yes, au courant du trouble, j'ai un guess de fix que j'essaie

File: include/server.hpp (hunk 1 above)
Comment (thelord82) [reply include/server.hpp:6]: Yes, they are necessary for opendir and readdir which are allowed functions in the new version of the subject that got bonified about a month ago.

**Pull request closed without merging**